truckersmp-cli

```diff
@@ -1,512 +1,547 @@
+
+def download_files(host, files_to_download):
+    try:
+        conn = http.client.HTTPSConnection(host)
+    except Exception as e:
+        printerr("Failed to connect to {}: {}".format(host, e))
+        return False
+    try:
+        while len(files_to_download) > 0:
+            path, dest, md5 = files_to_download[0]
+            md5hash = hashlib.md5()
+            bufsize = md5hash.block_size * 256
+            name = os.path.basename(path)
+            destdir = os.path.dirname(dest)
+            printdbg(
+              "Downloading file https://{}{} to {}".format(host, path, destdir))
+
+            # make file hierarchy
+            os.makedirs(destdir, exist_ok = True)
+
+            # download file
+            conn.request("GET", path, headers = {"Connection" : "keep-alive"})
+            res = conn.getresponse()
+
+            if res.status != 200:
+                printerr("Server response code is not 200.")
+                return False
+
+            lastmod = res.getheader("Last-Modified")
+            content_len = res.getheader("Content-Length")
+
+            with open(dest, "wb") as f:
+                downloaded = 0
+                while True:
+                    buf = res.read(bufsize)
+                    if not buf:
+                        break
+                    downloaded += len(buf)
+                    f.write(buf)
+                    md5hash.update(buf)
+                    if content_len:
+                        progress = "{:,} / {:,}".format(downloaded, int(content_len))
+                    else:
+                        progress = "{:,}".format(downloaded)
+                    print("\r{:40}{:>40}".format(name, progress), end = "")
+
+            if md5hash.hexdigest() != md5:
+                print("")  # newline
+                printerr("MD5 mismatch for {}".format(dest))
+                return False
+
+            # wget-like timestamping for downloaded files
+            if lastmod:
+                timestamp = time.mktime(
+                  time.strptime(lastmod, "%a, %d %b %Y %H:%M:%S GMT")) - time.timezone
+                try:
+                    os.utime(dest, (timestamp, timestamp))
+                except:
+                    pass
+
+            # downloaded successfully
+            print("\r{:40}{:>40}".format(name, "OK"))
+
+            # skip already downloaded files
+            # when trying to download from URL.dlurlalt
+            del files_to_download[0]
+    except Exception as e:
+        printerr("Failed to download https://{}{}: {}".format(host, path, e))
+        return False
+    finally:
+        conn.close()
+
+    return True
+
+def start_with_proton():
+    steamdir = None
+    # "steam" command in Debian package uses ~/.steam as steam user directory
+    # instead of $XDG_DATA_HOME/Steam
+    try:
+        with open("/usr/games/steam") as f:
+            for l in f:
+                if l.rstrip() == "config=$HOME/.steam":
```

Comment:
Is `/usr/games/steam` exclusive to debian ? It is not present on arch, but maybe other systems do something similar and instead of testing for `l == "config=$HOME/.steam"`, we could get the actual value from the `config=` field.

Reply:
Does Debian really use `~/.steam`? That's the old directory which is still there as fallback. There are only a few legacy files and symlinks to `$XDG_DATA_HOME/Steam` nowadays.

~~~
$ cat /usr/lib/steam/steam

# Get the default data path
	STEAM_DATA_HOME="${XDG_DATA_HOME:-$HOME/.local/share}"
	case "$STEAMPACKAGE" in
		steam)
			CLASSICSTEAMDIR="$HOME/Steam"
			DEFAULTSTEAMDIR="$STEAM_DATA_HOME/Steam"
~~~

Reply:
I asked a friend on Ubuntu how Steam was setup on her machine : `~/.steam` was an actual folder, not a symlink. `~/.local/share/steam` also existed, but was not used by steam.
That's anecdotal data, but it mean that a setup we might encounter.

Reply:
Though `$XDG_DATA_HOME/Steam` is not used on my Ubuntu 19.04, I removed the "detection" code for now. See #32. If we can add an option to specify the installation directory (such as `--steamdir DIR`), I'll add it.

Reply:
Now the script detects timestamp update in one of:
* `$XDG_DATA_HOME/Steam/config/loginusers.vdf`
* `~/.steam/config/loginusers.vdf`
* `~/.steam/debian-installation/config/loginusers.vdf`
and `inotifywait` is not used.

Reply:
But the maximum wait time is now 30 seconds? Steam takes longer for me to boot up completely and log me in.

Reply:
Changed to 99 seconds. I hope this is enough.